### PR TITLE
chore: Upgrade various dependencies

### PR DESCRIPTION
**Pull request by francislavoie**
Went through them one by one, some were renamed, some gained go.mod support, etc.

### CODE REVIEW

**Comment by francislavoie** [comment]
I remembered a random thing I wanted to do, while upgrading CEL. I wanted `file()` to work with no args. It's bit me a few times when helping people in the forums. So I fixed that.

File: modules/caddyhttp/fileserver/matcher.go [hunk 1]
```diff
@@ -199,7 +201,7 @@ func (MatchFile) CELLibrary(ctx caddy.Context) (cel.Library, error) {
 		cel.Function("file", cel.Overload("file_request_map", []*cel.Type{requestType, caddyhttp.CELTypeJSON}, cel.BoolType)),
 		cel.Function("file_request_map",
 			cel.Overload("file_request_map", []*cel.Type{requestType, caddyhttp.CELTypeJSON}, cel.BoolType),
-			cel.SingletonBinaryImpl(caddyhttp.CELMatcherRuntimeFunction("file_request_map", matcherFactory))),
+			cel.SingletonBinaryBinding(caddyhttp.CELMatcherRuntimeFunction("file_request_map", matcherFactory))),
```
Comment (francislavoie) [inline modules/caddyhttp/fileserver/matcher.go:204]: This function was deprecated/renamed

File: modules/caddyhttp/fileserver/matcher.go [hunk 2]
```diff
@@ -245,11 +250,11 @@ func celFileMatcherMacroExpander() parser.MacroExpander {
...
-			eh.NewMap(
-				eh.NewMapEntry(
-					eh.LiteralString("try_files"), eh.NewList(args...),
-				),
-			),
+			eh.NewMap(eh.NewMapEntry(
+				eh.LiteralString("try_files"),
+				eh.NewList(args...),
+				false,
+			)),
```
Comment (francislavoie) [inline modules/caddyhttp/fileserver/matcher.go:257]: `NewMapEntry` gained a new 3rd arg `optional bool`. I set it to false.

File: modules/caddyhttp/tracing/tracer.go [hunk 3]
```diff
@@ -108,8 +108,8 @@ func (ot *openTelemetryWrapper) newResource(
 	webEngineVersion string,
 ) (*resource.Resource, error) {
 	return resource.Merge(resource.Default(), resource.NewSchemaless(
-		semconv.WebEngineNameKey.String(webEngineName),
-		semconv.WebEngineVersionKey.String(webEngineVersion),
+		semconv.WebEngineName(webEngineName),
+		semconv.WebEngineVersion(webEngineVersion),
```
Comment (francislavoie) [inline modules/caddyhttp/tracing/tracer.go:112]: The old way is still functional, but I spotted a slight API improvement in `v1.13.0` so I made the change.

File: modules/caddyhttp/app.go [hunk 4]
```diff
@@ -54,16 +54,17 @@ func init() {
...
 // `{http.request.header.*}` | Specific request header field
-// `{http.request.host.labels.*}` | Request host labels (0-based from right); e.g. for foo.example.com: 0=com, 1=example, 2=foo
 // `{http.request.host}` | The host part of the request's Host header
+// `{http.request.host.labels.*}` | Request host labels (0-based from right); e.g. for foo.example.com: 0=com, 1=example, 2=foo
 // `{http.request.hostport}` | The host and port from the request's Host header
 // `{http.request.method}` | The request method
 // `{http.request.orig_method}` | The request's original method
+// `{http.request.orig_uri}` | The request's original URI
+// `{http.request.orig_uri.path}` | The request's original path
+// `{http.request.orig_uri.path.*}` | Parts of the original path, split by `/` (0-based from left)
```
Comment (francislavoie) [inline modules/caddyhttp/app.go:64]: It was missing in the godoc after https://github.com/caddyserver/caddy/pull/5161

**Review by mholt — APPROVED** [review]
LGTM -- thanks for your effort on this!

**Pull request merged**